### PR TITLE
General cleanup and refactoring of Diagnostics Common and AspNetCore Trace integration tests and snippets.

### PR DESCRIPTION
**Pull request by amanda-tarafa**
This is done mainly to reuse code in some upcoming PRs which will add AspNet Trace integration tests and snippets.

Minor refactor to one of the ErrorReporting test that slipped refactoring done in #2372

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/SimpleManagedTracerTest.cs [hunk 1]
```diff
@@ -270,5 +231,25 @@ public void Trace_IncompleteSpans()
...
+            DateTime now = DateTime.UtcNow;
+            while (now >= DateTime.UtcNow)
+            {
+                Thread.Sleep(1);
+            }
+        }
+
+        public void Dispose()
+        {
+            _grpcConsumer.Dispose();
```
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/SimpleManagedTracerTest.cs]: suggestion: Dispose() => _grpcConsumer.Dispose();

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/TraceEntryVerifiers.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+// limitations under the License.
+
+using Google.Cloud.Trace.V1;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+using TraceProto = Google.Cloud.Trace.V1.Trace;
+
+namespace Google.Cloud.Diagnostics.Common.IntegrationTests
```
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/TraceEntryVerifiers.cs]: Very nice clean up!

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/TraceEntryVerifiers.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            Assert.Equal(expectedLabels.Count, span.Labels.Count);
+            AssertSpanLabelsContains(span, expectedLabels);
+        }
+
+        /// <summary>
+        /// Verifies that the span contains the set of expected labels.
+        /// The span can have other labels.
+        /// </list>
+        /// </summary>
+        /// <param name="span">The spna to check.</param>
```
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/TraceEntryVerifiers.cs]: nit: span

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/SimpleManagedTracerTest.cs (hunk 1 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/SimpleManagedTracerTest.cs]: Done!

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/TraceEntryVerifiers.cs (hunk 2 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/TraceEntryVerifiers.cs]: Thanks :)

File: apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/TraceEntryVerifiers.cs (hunk 3 above)
Comment (amanda-tarafa) [reply apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.IntegrationTests/Trace/TraceEntryVerifiers.cs]: Done

**Comment by amanda-tarafa** [comment]
Thanks!

**Pull request merged**